app/src/main/java/com/zulip/android/models/Stream.java

```diff
@@ -179,6 +180,32 @@ public static Stream getByName(ZulipApp app, String name) {
         return stream;
     }
 
+    /**
+     * Checks stream name is valid or not
+     * @param app ZulipApp
+     * @param streamName Checks this stream name is valid or not
+     * @return null if stream does not exist else cursor
+     */
+    public static List<Stream> streamCheckBeforeMessageSend(ZulipApp app, CharSequence streamName) {
+        if (streamName == null) {
+            return null;
+        }
+        try {
+            List<Stream> list = app.getDao(Stream.class)
+                    .queryBuilder().where()
+                    .eq(Stream.NAME_FIELD, new SelectArg(Stream.NAME_FIELD, streamName)).query();
```

Comment:
Nice! this is even better :) Also you can even you queryForFirst() to get the first result straightway.